hal/include/hal_types.h

```diff
@@ -169,7 +169,7 @@ struct hal_device_info_t {
   uint32_t max_workgroup_size;
 
   /// @brief true if little endian.
-  bool is_little_endian;
+  bool is_little_endian = true;
```

Comment:
We should be detecting this rather than assuming, I think.

Reply:
This is a default in the base Hal, it's up to the writer of each HAL to dictate this value (also because clc uses this an offline way, we can't easily detect other than the host) - I have added some additional checks on the host side though to see that it has the same reported endianness.

Reply:
That makes sense, being unable to detect it. But it still shouldn't be necessary to set it here. I see it's already getting set as it should be in `cpu_hal::setup_cpu_hal_device_info`.

Reply:
This is more about avoiding uninitialized variables than it being the defacto value.

Reply:
But it gets initialized to `true` rather than `{}`, and other fields are not initialized similarly.

Reply:
fair enough, I'll remove it, I've just seen issues with not setting things to default values here, especially when we add fields, but I'll think on it and remove it from this change.

Reply:
removed.